src/compiler/c.rs

```diff
@@ -445,6 +561,379 @@ where
     }
 }
 
+const PRAGMA_GCC_PCH_PREPROCESS: &[u8] = b"pragma GCC pch_preprocess";
+const HASH_31_COMMAND_LINE_NEWLINE: &[u8] = b"# 31 \"<command-line>\"\n";
+const HASH_32_COMMAND_LINE_2_NEWLINE: &[u8] = b"# 32 \"<command-line>\" 2\n";
+const INCBIN_DIRECTIVE: &[u8] = b".incbin";
+
+/// Remember the include files in the preprocessor output if it can be cached.
+/// Returns `false` if direct mode should be disabled.
+fn process_preprocessed_file(
```

Comment:
i feel this could be a tested a bit better?
maybe by splitting it into smaller function?

Reply:
I'll try splitting into smaller functions first and seeing what kind of unit tests I can come up with.

Reply:
Hopefully this change is what you had in mind.